pkg/scm/bitbucket.go

```diff
@@ -0,0 +1,249 @@
+	if err != nil {
+		return Context{}, err
+	}
+	emailAddress := ""
+	emailValues, ok := emails.(map[string]interface{})
+	if !ok {
+		return Context{}, fmt.Errorf(emailError)
+	}
+	emailAddress, err = getEmailAddress(emailValues)
+	if err != nil {
+		return Context{}, err
+	}
+
+	wl, err := b.Client.Workspaces.List()
+	if err != nil {
+		return Context{}, err
+	}
+
+	workspaces := make([]string, 0)
+	for _, w := range wl.Workspaces {
+		workspaces = append(workspaces, w.Slug)
+	}
+
+	if len(workspaces) == 0 {
+		return Context{}, fmt.Errorf("You don't have any Bitbucket workspace created. Please create one first \n")
+	}
+
+	workspace := workspaces[0]
+	if len(workspaces) > 1 {
+		prompt := &survey.Select{
+			Message: "Select the workspace:",
+			Options: workspaces,
+		}
+		if err := survey.AskOne(prompt, &workspace, survey.WithValidator(survey.Required)); err != nil {
+			return Context{}, err
+		}
+	}
+
+	pr, err := b.Client.Workspaces.Projects(workspace)
+	if err != nil {
+		return Context{}, err
+	}
+	projectKeys := make(map[string]string, 0)
+	projects := make([]string, 0)
+	for _, p := range pr.Items {
+		projectKeys[p.Name] = p.Key
+		projects = append(projects, p.Name)
+	}
+
+	project := ""
+	if len(projects) == 0 {
+		return Context{}, fmt.Errorf("You don't have any Bitbucket project created. Please create one first \n")
+	}
+
+	prompt := &survey.Select{
+		Message: "Select the project for your repo:",
+		Options: projects,
+	}
+	if err := survey.AskOne(prompt, &project, survey.WithValidator(survey.Required)); err != nil {
+		return Context{}, err
+	}
+
+	pub, priv, err := GenerateKeys(false)
+	if err != nil {
+		return Context{}, err
+	}
+
+	repoName, err := repoName()
+	if err != nil {
+		return Context{}, err
+	}
+
+	opt := &bitbucket.RepositoryOptions{
+		Uuid:      "",
+		Owner:     user.Username,
+		RepoSlug:  repoName,
+		Scm:       "git",
+		IsPrivate: "true",
+		Project:   projectKeys[project],
+	}
+
+	utils.Highlight("\ncreating bitbucket repository %s...\n", repoName)
+
+	res, err := b.Client.Repositories.Repository.Create(opt)
+	if err != nil {
+		return Context{}, err
+	}
+	sshAddress, err := getSSHAddress(res.Links)
+	if err != nil {
+		return Context{}, err
+	}
+
+	utils.Highlight("Setting up a read-write deploy key for this repo...\n")
+
+	if _, err := b.Client.Repositories.DeployKeys.Create(&bitbucket.DeployKeyOptions{
+		Owner:    user.Username,
+		RepoSlug: repoName,
+		Label:    "Plural Deploy Key",
+		Key:      pub,
+	}); err != nil {
+		return Context{}, err
+	}
+	// The bitbucket creates empty repository.
+	// Initialize bitbucket repo with .gitignore file.
+	dir, err := os.MkdirTemp("", "bitbucket")
+	if err != nil {
+		return Context{}, err
+	}
+	defer os.RemoveAll(dir)
+	gitIgnore := path.Join(dir, ".gitignore")
+	if err := os.WriteFile(gitIgnore, []byte(""), 0644); err != nil {
+		return Context{}, err
+	}
+
+	if err := b.Client.Repositories.Repository.WriteFileBlob(&bitbucket.RepositoryBlobWriteOptions{
```

Comment:
why do we need to do this?  `plural init` should override the gitignore (i remember another bitbucket user hitting this snag a looong time ago)

Reply:
Is there no api that creates the repo with an initial commit?

Reply:
I feel like i've seen users have an initial commit, maybe it's set at a different level in bitbucket

Reply:
there is no API for this. Only UI has the option to create README and .gitignore files